### DISCUSSION TITLE
Make bucket public

### DISCUSSION THREAD

**Discussion by otecmasters** [supabase/supabase#16788] [Questions] [answered]
Hello, just like I can disable the rls through a migration in sql. Is there a command that allows me to leave a public bucket? To then upload those changes and see them reflected in develop and production? Or can I change them directly without affecting future merges?

for example i used this to disable a rls
```
alter table public.stages
    disable row level security;
```

**Comment by GaryAustin1** [comment, marked as answer]
From the docs on storage you can create a bucket with SQL.  Public is just a column in that table.
```
insert into storage.buckets
  (id, name,public)
values
  ('avatars', 'avatars',true);
  ```
You can also change an existing bucket my just updating the row.

You can set RLS policies thru SQL on storage.objects.  You should not disable RLS on storage schema though as you should never modify Supabase schemas like auth and storage.

EDIT: WARNING a Public bucket only allows a public URL access.  It does not allow uploads and other API calls to work.   Those are managed by RLS.

**Reply by otecmasters** [reply]
hi!, 
Thanks for the answer. Yes i have change the bucket from public to private because i couldn't get the publicUrl with the function `getPublicUrl` because it was not public. And for the RLS policies i have to adjust the policies for those tables.
